pkg/blob/controllerserver.go

```diff
@@ -802,6 +803,33 @@ func (d *Driver) copyBlobContainer(ctx context.Context, req *csi.CreateVolumeReq
 			return fmt.Errorf("timeout waiting for copy blob container %s to %s complete, current copy percent: %s%%", srcContainerName, dstContainerName, percent)
 		}
 		copyErr := util.WaitUntilTimeout(time.Duration(d.waitForAzCopyTimeoutMinutes)*time.Minute, execFunc, timeoutFunc)
+		if dstAccountSasToken == "" && copyErr != nil && strings.Contains(copyErr.Error(), authorizationPermissionMismatch) {
+			klog.Warningf("azcopy copy failed with AuthorizationPermissionMismatch error, should assign \"Storage Blob Data Contributor\" role to controller identity, fall back to use sas token, original error: %v", copyErr)
+			var srcSasToken, dstSasToken string
```

Comment:
pls use `copyBlobContainer` to avoid duplicate codes, you could call `copyBlobContainer` func twice: 1) use default auth 2) if returns with `authorizationPermissionMismatch` then call `copyBlobContainer` func again to use sas token by force

Reply:
done, use this logic in `copyVolume`